packages/amplify-category-api/src/graphql-transformer/override.ts

```diff
@@ -63,26 +62,17 @@ export function applyFileBasedOverride(stackManager: StackManager, overrideDirPa
   });
 
   const appsyncResourceObj = convertToAppsyncResourceObj(amplifyApiObj);
-  const overrideCode: string = fs.readFileSync(overrideFilePath, 'utf-8');
-  const sandboxNode = new vm.NodeVM({
-    console: 'inherit',
-    timeout: 5000,
-    sandbox: {},
-    require: {
-      context: 'sandbox',
-      builtin: ['path'],
-      external: true,
-    },
-  });
-  // Remove these when moving override up to amplify-category-api level
   const { envName } = stateManager.getLocalEnvInfo();
   const { projectName } = stateManager.getProjectConfig();
   const projectInfo = {
     envName,
     projectName,
   };
   try {
-    sandboxNode.run(overrideCode, overrideFilePath).override(appsyncResourceObj, projectInfo);
+    const overrideImport = require(overrideFilePath);
```

Comment:
Since this is a synchronous method, await import doesn't work. Have to fall back to require.

Reply:
How bad is making call stack async ?

Reply:
pretty intense. It ties into the `transform` and thereby `compile`. Will need an API change as well.

Reply:
That'd be problematic until this refactor https://github.com/aws-amplify/amplify-category-api/pull/1699 is complete (and won't be done for a week probably). We can revisit at that point, but until then, this OverrideCallback is in the CDK construct ctor critical path :(

Reply:
Sounds good.
Lesson learnt - going forward it's good to skew towards async apis earlier.
I.e. if there's any fs access - use async apis not sync as forcing factor.